pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -639,12 +639,10 @@ func shouldPodBeInEndpoints(pod *v1.Pod) bool {
 
 func endpointPortFromServicePort(servicePort *v1.ServicePort, portNum int) *v1.EndpointPort {
 	epp := &v1.EndpointPort{
-		Name:     servicePort.Name,
-		Port:     int32(portNum),
-		Protocol: servicePort.Protocol,
-	}
-	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAppProtocol) {
-		epp.AppProtocol = servicePort.AppProtocol
+		Name:        servicePort.Name,
+		Port:        int32(portNum),
+		Protocol:    servicePort.Protocol,
+		AppProtocol: servicePort.AppProtocol,
```

Comment:
Do we not have this equivalent in EPSlice?  Why not?

Reply:
Good question - it actually predated all of this (v1.17). This feature gate guards a later addition of the same field to Services and Endpoints: https://github.com/kubernetes/kubernetes/blob/release-1.17/staging/src/k8s.io/api/discovery/v1alpha1/types.go#L142-L148.